src/Extension/FluentVersionedExtension.php

```diff
@@ -329,4 +357,70 @@ public function flushCache()
     {
         $this->localisedStageCache = [];
     }
+
+    /**
+     * Hook into {@link Hierarchy::prepopulateTreeDataCache}.
+     *
+     * @param DataList $recordList The list of records to prepopulate caches for. Null for all records.
+     * @param array $options A map of hints about what should be cached. "numChildrenMethod" and
+     *                       "childrenMethod" are allowed keys.
+     */
+    public function onPrepopulateTreeDataCache(DataList $recordList = null, array $options = [])
+    {
+        if (!Config::inst()->get(static::class, 'prepopulate_localecontent_cache')) {
+            return;
+        }
+
+        // This functionality hasn't been implemented; it's not actually used right now, so assume you aren't going to
+        // need it.
+        if ($recordList) {
+            user_error(
+                "FluentVersionedExtension::onPrepopulateTreeDataCache not currently optimised for recordList use",
+                E_USER_WARNING
+            );
+        }
+
+        self::prepoulateIdsInLocale(FluentState::singleton()->getLocale(), $this->owner->baseClass());
+    }
+
+    /**
+     * Prepopulate the cache of IDs in a locale, to optimise batch calls to isLocalisedInStage.
+     *
+     * @param string $locale
+     * @param string $dataObject
+     */
+    public static function prepoulateIdsInLocale($locale, $dataObjectClass, $populateLive = true, $populateDraft = true)
+    {
+        // Get the table for the given DataObject class
+        $self = new static();
+        $table = $self->getLocalisedTable(Injector::inst()->get($dataObjectClass)->baseTable());
```

Comment:
Wouldn't you be better off getting a singleton of `$dataObjectClass` and calling `getLocalisedTable` on that.

```php
$singleton = DataObject::singleton($dataObjectClass);
$table = $singleton->getLocalisedTable();
```

Reply:
I can use the `singleton` method :+1: - I'll also grab a singleton of the extension instead... I don't think the `DataObjectSchema` method will give the same result as there's some additional logic in `FluentExtension::getLocalisedTable`?